functests-extended/1_performance_operator_upgrade/upgrade_operator.go

```diff
@@ -17,45 +18,52 @@ import (
+	// TODO add new test id for 4.5 -> 4.6 upgrade
+	It("[test_id:unknown] upgrades performance profile operator", func() {
 		operatorNamespace := "openshift-performance-addon"
 		subscriptionName := "performance-addon-operator"
-		previousVersion := "4.4.0"
-		currentVersion := "4.5.0"
 
-		By(fmt.Sprintf("Verifying that %s channel is active", previousVersion))
+		Expect(fromVersion).ToNot(BeEmpty(), "fromVersion not set")
+		Expect(toVersion).ToNot(BeEmpty(), "toVersion not set")
+
+		By(fmt.Sprintf("Upgrading from %s to %s", fromVersion, toVersion))
+
+		By(fmt.Sprintf("Verifying that %s channel is active", fromVersion))
 		subscription := getSubscription(subscriptionName, operatorNamespace)
-		Expect(subscription.Spec.Channel).To(Equal(previousVersion))
-		Expect(subscription.Status.CurrentCSV).To(ContainSubstring(previousVersion))
+		Expect(subscription.Spec.Channel).To(Equal(fromVersion))
+		Expect(subscription.Status.CurrentCSV).To(ContainSubstring(fromVersion))
 
 		// CSV is pointed to previous image tag and CRD is an old version
 		csv := getCSV(subscription.Status.CurrentCSV, operatorNamespace)
-		Expect(csv.ObjectMeta.Annotations["containerImage"]).To(ContainSubstring(previousVersion))
-
-		crd := getCRD(csv.Spec.CustomResourceDefinitions.Owned[0].Name)
-		Expect(crd.Spec.Validation.OpenAPIV3Schema).NotTo(ContainSubstring("topologyPolicy"))
+		// channel 4.y.z might still use snapshot image 4.y-snapshot, so only major and minor version will match.
+		fromMajorMinor := string([]rune(fromVersion)[0:3])
+		Expect(csv.ObjectMeta.Annotations["containerImage"]).To(ContainSubstring(fromMajorMinor))
 
-		By(fmt.Sprintf("Switch subscription channel to %s version", currentVersion))
+		By(fmt.Sprintf("Switch subscription channel to %s version", toVersion))
 		Expect(testclient.Client.Patch(context.TODO(), subscription,
 			client.ConstantPatch(
 				types.JSONPatchType,
-				[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/channel", "value": "%s" }]`, currentVersion)),
+				[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/channel", "value": "%s" }]`, toVersion)),
 			),
 		)).ToNot(HaveOccurred())
 
-		By(fmt.Sprintf("Verifying that channel was updated to %s", currentVersion))
-		subscriptionWaitForUpdate(subscription.Name, operatorNamespace, currentVersion)
+		By(fmt.Sprintf("Verifying that channel was updated to %s", toVersion))
+		subscriptionWaitForUpdate(subscription.Name, operatorNamespace, toVersion)
 
 		// CSV is updated and pointed to right image tag
 		subscription = getSubscription(subscriptionName, operatorNamespace)
 		csv = getCSV(subscription.Status.CurrentCSV, operatorNamespace)
 		csvWaitForPhaseWithConditionReason(csv.Name, operatorNamespace, olmv1alpha1.CSVPhaseSucceeded, olmv1alpha1.CSVReasonInstallSuccessful)
-
-		// CRD should be current version with all new features
-		crd = getCRD(csv.Spec.CustomResourceDefinitions.Owned[0].Name)
```

Comment:
hm, as I see we don't have CRD verification.. I'm not sure we are really ok with that, but if we don't need it - we can remove unused function: getCRD()

Reply:
There is nothing we can test yet, 4.5 and 4.6 CRDs are the same AFAIK.
Removed unused func, added test id

Reply:
proably it should be included in https://github.com/openshift-kni/performance-addon-operators/pull/249 or in a following PR after merge.